src/cloudai/_core/base_runner.py

```diff
@@ -288,17 +288,22 @@ async def monitor_jobs(self) -> int:
                 if self.mode == "dry-run":
                     successful_jobs_count += 1
                 else:
-                    job_status_result = self.get_job_status(job)
-                    if job_status_result.is_successful:
+                    if job.test.job_status_check:
+                        job_status_result = self.get_job_status(job)
+                        if job_status_result.is_successful:
+                            successful_jobs_count += 1
+                            await self.handle_job_completion(job)
+                        else:
+                            error_message = (
+                                f"Job {job.id} for test {job.test.section_name} failed: "
+                                f"{job_status_result.error_message}"
+                            )
+                            logging.error(error_message)
+                            await self.shutdown()
+                            raise JobFailureError(job.test.section_name, error_message, job_status_result.error_message)
+                    else:
```

Comment:
@TaekyungHeo  Here, can we add the job status even if we choose to ignore the failure and proceed? When running it in NVIS flow, even though it prevents an auto-exit on failure, it says "Job completed". I think capturing the error and logging would avoid that confusion.

```
else:
                        job_status_result = self.get_job_status(job)
                        if not job_status_result.is_successful:
                            error_message = (
                                    f"Job {job.id} for test {job.test.section_name} failed: {job_status_result.error_message}"
                                )
                            logging.error(error_message)
                        successful_jobs_count += 1
                        await self.handle_job_completion(job)
```